sno/dataset2.py

```diff
@@ -593,3 +618,17 @@ def import_iter_feature_blobs(self, resultset, source):
         for feature in resultset:
             feature_path, feature_content = self.encode_feature(feature, schema)
             yield self.repo_path(feature_path), feature_content
+
+    @property
+    def primary_key(self):
+        # TODO - datasets v2 model supports more than one primary key.
+        # This function needs to be changed when we have a working copy v2 that does too.
+        if len(self.schema.pk_columns) == 1:
+            return self.schema.pk_columns[0].name
+        raise ValueError(f"No single primary key: {self.schema.pk_columns}")
+
+    def encode_feature_blob(self, feature):
+        # TODO - the dataset interface still needs some work:
+        # - maybe encode methods shouldn't be classmethods.
+        # - having a _blob version of encode_feature is too many similar methods.
```

Comment:
Best approach is probably to profile various dataset operations (checkout/import/diff) and fix the hotspots — that's what I did first time around and why there was a proliferation of similar-but-not-quite methods. 

I'd hope our new structure is slightly cleaner and they're maybe needed less.

Reply:
This particular method will hopefully go away if we fix diffs to not manually compare blob hashes for rename detection, and leave that to pygit2, since that is the only caller.